### ISSUE TITLE
Failed assertion when parsing [%bs.obj]

### ISSUE THREAD

**Issue by ryb73** [facebook/reason#1222]
I got the following error when running `refmt` with no options:

```
refmt: internal error, uncaught exception:
       File "src/reason_pprint_ast.ml", line 4969, characters 13-19: Assertion failed
```

My input file:
```ocaml
type pathObject =
  [%bs.obj: <
         dir : string ;
         root : string ;
         base : string ;
         name : string ;
         ext : string          
       >            
  ] 
```

I'm using Reason 1.13.0.

**Comment by bsansouci** [comment]
Hey @ryb73 I think the types here would be simply
```
type pathObject = <
         dir : string ;
         root : string ;
         base : string ;
         name : string ;
         ext : string          
       >
```

which actually today's syntax in Reason is:

```
type pathObject = {. /* The dot means it's a close object type. You can use 2 dots to mean open object type. */
  dir : string, 
  root : string, 
  base : string, 
  name : string,
  ext : string
};
```

What you use `%bs.obj` for is the literal, the value of the object:
```
let a: pathObject = [%bs.obj {dir: "asd"}]
/* actually we have premium syntax for those JS objects, which is the following */
let a: pathObject = {"dir": "asd"} /* notice the quotes around the key */
```

I'd recommend using [reason-tools](https://github.com/reasonml/reason-tools/) to look at what all those become into OCaml if that helps. Or compile to JS to see the output.

**Comment by Schmavery** [comment]
@bsansouci actually it should be possible to use %bs.obj for both types and values: https://bloomberg.github.io/bucklescript/Manual.html#_create_js_objects_using_bs_obj

**Comment by chenglou** [comment]
Fixed by #1659. There's a better warning for this now
